codex/sales/reservations.nim

```diff
@@ -467,7 +485,7 @@ proc storables(
       return failure(error)
   elif T is Reservation:
     # should indicate key length of 5, but let the .key logic determine it
-    without defaultKey =? key(ReservationId.default, AvailabilityId.default), error:
+    without defaultKey =? key(ReservationId.default, availabilityId |? AvailabilityId.default), error:
```

Comment:
The addition of the `availabilityId` is probably not good for this routine. Below, we're simply comparing the number of namespaces for the key, which is different for Availabilities and Reservations (and admittedly a weak design because the namespaces can change). By specifying the availabiltyId, we're not changing the query logic to find reservations for a particular availability. 

Instead, I think better options might include either:
1. pass a filter predicate into this routine to add to the `next` query filtering, OR
2. query all Reservations then filter by AvailabilityId (in memory)